bpf/bpf_host.c

```diff
@@ -866,6 +861,15 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, const bool from_host)
 		magic = inherit_identity_from_host(ctx, &identity);
 		if (magic == MARK_MAGIC_PROXY_INGRESS ||  magic == MARK_MAGIC_PROXY_EGRESS)
 			trace = TRACE_FROM_PROXY;
+
+#ifdef ENABLE_IPSEC
+		if (magic == MARK_MAGIC_ENCRYPT) {
+			send_trace_notify(ctx, trace, identity, 0, 0, ctx->ingress_ifindex,
```

Comment:
Yeah, during looking back at this PR, I found even `FROM_HOST` is not always the case. For example, if endpoint routing is enabled, there is a possible path of native interface ingress -> cilium_host egress. I'll raise it up in the datapath channel.

from-network (bpf_network) -> do_decrypt
https://github.com/cilium/cilium/blob/d3742a6e018ca6391d3cc9a7d7c1dea367b64d90/bpf/lib/encrypt.h#L70-L74

Reply:
Yep, agree. But unfortunately, because of the endpoint routing, it is not always the case that the packet comes from the host/stack. I think it is impossible to know where it comes from here.

Reply:
Yep, agree. I just copied this from the current code. On the other hand, there is another problem here I think. If the endpoint route is disabled, it may come from a network device to from-host directly. In this case, using `from-network` should be more suitable.

How about leaving this problem as-is for now and thinking about the problem of `obs_point` correctness in the different issues?

Reply:
> If the endpoint route is disabled, it may come from a network device to from-host directly (https://github.com/cilium/cilium/blob/master/bpf/lib/encrypt.h#L70-L74). In this case, using `from-network` should be more suitable.

But in that case, we will have cleared the packet mark before (https://github.com/cilium/cilium/blob/v1.11.1/bpf/lib/encrypt.h#L69) so we won't hit this `magic == MARK_MAGIC_ENCRYPT` case here, no?

> How about leaving this problem as-is for now and thinking about the problem of `obs_point` correctness in the different issues?

The trace did not exist before and will be added by this commit. I think it's preferable to not have a packet trace rather than introduce a potentially confusing one. So I'd rather we address this concern now.

Reply:
But in that case, we will have cleared the packet mark before (https://github.com/cilium/cilium/blob/v1.11.1/bpf/lib/encrypt.h#L69) so we won't hit this magic == MARK_MAGIC_ENCRYPT case here, no?

Ah, yes. You are right. It clears the mark, and also bpf_network is only relevant with the decryption path, so never comes with MARK_MAGIC_ENCRYPT. Sorry 🙇

Reply:
With some conversation out of GitHub, we concluded that we should use `TRACE_FROM_STACK`. I'll fix the commit to use that.